### PR TITLE
Zeroconf tweaks and error handling

### LINKED ISSUE #6077: zeroconf: 'NoneType' object has no attribute 'address'
### Observed behavior

unhandled error while starting up

### Expected behavior

no unhandled errors

### User-facing consequences

unknown

### Errors and logs

```
Exception in thread zeroconf-ServiceBrowser_Kolibri._sub._http._tcp.local.:
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/Users/d/.pex/install/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl.2a207c91c66d601df134ab0be317299b564f33c9/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl/kolibri/dist/zeroconf.py", line 1442, in run
    handler(self.zc)
  File "/Users/d/.pex/install/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl.2a207c91c66d601df134ab0be317299b564f33c9/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl/kolibri/dist/zeroconf.py", line 1383, in <lambda>
    state_change=state_change,
  File "/Users/d/.pex/install/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl.2a207c91c66d601df134ab0be317299b564f33c9/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl/kolibri/dist/zeroconf.py", line 1279, in fire
    h(**kwargs)
  File "/Users/d/.pex/install/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl.2a207c91c66d601df134ab0be317299b564f33c9/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl/kolibri/dist/zeroconf.py", line 1352, in on_change
    listener.add_service(*args)
  File "/Users/d/.pex/install/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl.2a207c91c66d601df134ab0be317299b564f33c9/kolibri-0.13.0.dev0+git.20191116091820-py2.py3-none-any.whl/kolibri/core/discovery/utils/network/search.py", line 108, in add_service
    ip = socket.inet_ntoa(info.address)
AttributeError: 'NoneType' object has no attribute 'address'
```

### Steps to reproduce

ran a pex from the latest `develop`

### Context

0.12.0 alpha 0

### LINKED ISSUE #6390: unhandled zeroconf exception on shutdown
### Observed behavior

when shutting down Kolibri, I see unhandled errors from zeroconf processes.

```
(k013x) d@slate ~/P/l/k013x (release-v0.13.x)> kolibri start
WARNING:root:No C Extensions available for this platform.

INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Running Kolibri
INFO     Kolibri running on:

	http://172.27.5.138:8080/
	http://127.0.0.1:8080/

INFO     Starting Kolibri 0.13.0b7.dev0+git.116.g19a41325
INFO     ENGINE Listening for SIGHUP.
INFO     ENGINE Listening for SIGUSR1.
INFO     ENGINE Listening for SIGTERM.
INFO     ENGINE Listening for SIGINT.
INFO     ENGINE Bus STARTING
INFO     Started new JOBCHECKER thread ID#a095c6d41ff2434d8c88a09f205483f4
INFO     Started new JOBCHECKER thread ID#a6d339d9adfa4b398e3cd4c15afb9c70
INFO     Started new SCHEDULECHECKER thread ID#e0c08d99b51f4946ac02c008d6c0eddc
INFO     Registering ourselves to zeroconf network with id '4d7f0be7ea562102062ec84450a1d357'...
INFO     ENGINE Serving on http://0.0.0.0:8080
INFO     ENGINE Bus STARTED
INFO     Ping succeeded! (response: {u'messages': [], u'id': 513065})
^[[1;5INFO     127.0.0.1 - - "GET /static/kolibri.core.default_frontend/kolibri.core.default_frontend0.13.0b7.dev0+git.116.g19a41325.css HTTP/1.1" 404 489 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/assets/fonts/noto-subset.en.css?v=0.13.0b7.dev0+git.116.g19a41325 HTTP/1.1" 200 16039 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.plugins.facility.app/kolibri.plugins.facility.app0.13.0b7.dev0+git.116.g19a41325.css HTTP/1.1" 404 488 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.core.default_frontend/kolibri.core.default_frontend-0.13.0b7.dev0+git.116.g19a41325.js HTTP/1.1" 404 488 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.plugins.facility.app/kolibri.plugins.facility.app-0.13.0b7.dev0+git.116.g19a41325.js HTTP/1.1" 404 489 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/assets/fonts/noto-common.css?v=0.13.0b7.dev0+git.116.g19a41325 HTTP/1.1" 200 108608 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.core.default_frontend/kolibri.core.default_frontend0.13.0b7.dev0+git.116.g19a41325.css HTTP/1.1" 404 489 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/assets/fonts/noto-full.en.modern.css?v=0.13.0b7.dev0+git.116.g19a41325 HTTP/1.1" 200 2549 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.core.default_frontend/kolibri.core.default_frontend-0.13.0b7.dev0+git.116.g19a41325.js HTTP/1.1" 404 488 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.plugins.facility.app/kolibri.plugins.facility.app0.13.0b7.dev0+git.116.g19a41325.css HTTP/1.1" 404 488 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
INFO     127.0.0.1 - - "GET /static/kolibri.plugins.facility.app/kolibri.plugins.facility.app-0.13.0b7.dev0+git.116.g19a41325.js HTTP/1.1" 404 489 "http://localhost:8080/en/facility/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36"
^CINFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
^CINFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) already shut down
WARNING  SCHEDULECHECKER shut down event received; closing.
INFO     
Kolibri instance '4d7f0be7ea562102062ec84450a1d357' has left the zeroconf network.

WARNING  JOBCHECKER shut down event received; closing.
WARNING  JOBCHECKER shut down event received; closing.
INFO     ENGINE Bus STOPPED
INFO     ENGINE Bus EXITING
INFO     ENGINE Bus EXITED
INFO     ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
INFO     ENGINE Bus STOPPED
INFO     ENGINE Bus EXITING
INFO     ENGINE Bus EXITED
INFO     ENGINE Waiting for child threads to terminate...
Exception in thread expire_cache (most likely raised during interpreter shutdown):Exception in thread zeroconf-Reaper (most likely raised during interpreter shutdown):

Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
  File "/Users/d/Projects/le/k013x/.venv/lib/python2.7/site-packages/cherrypy/lib/caching.py", line 199, in expire_cache
<type 'exceptions.TypeError'>: 'NoneType' object is not callable
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
  File "/Users/d/Projects/le/k013x/.venv/lib/python2.7/site-packages/zeroconf.py", line 1264, in run
  File "/Users/d/Projects/le/k013x/.venv/lib/python2.7/site-packages/zeroconf.py", line 1838, in wait
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 289, in __exit__
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 216, in __exit__
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 203, in release
<type 'exceptions.TypeError'>: 'NoneType' object is not callable
```

### Expected behavior

no unhandled errors

### User-facing consequences

unknown

### Steps to reproduce

* Start Kolibri
* Shut it down

### Context

0.13.0 beta, after #6380 was merged

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Adds handling for a `get_service_info` timeout.
* Cleans up the Zeroconf thread before shutdown
* Removes bare asserts in zeroconf methods
* Cleans up logging statements to remove unnecessary line breaks

### Reviewer guidance
Run the server make sure everything looks dandy.
None of the changes should change the API we need to expose to zeroconf, but good to just double check!

### References
Fixes #6077
Fixes #6390

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/discovery/utils/network/search.py [hunk 1]
```diff
@@ -88,7 +87,9 @@ def register(self):
 
     def unregister(self):
 
-        assert self.info is not None, "Service is not registered!"
+        if self.info is None:
+            logging.error("Service is not registered!")
+            return
```
Comment (indirectlylit) [inline kolibri/core/discovery/utils/network/search.py:92]: nice. much better than `assert`

File: kolibri/core/discovery/utils/network/search.py [hunk 2]
```diff
@@ -140,27 +149,30 @@ def add_service(self, zeroconf, type, name):
...
-                        + "service info:\n %s;\n\nThe following exception was raised:\n%s"
-                    )
+                    """
+                        A new Kolibri instance '%s' was seen on the zeroconf network,
+                        but we had trouble getting the information we needed about it.
+                        Service info:
+                        %s
+                        The following exception was raised:
+                        %s
+                        """
```
Comment (indirectlylit) [inline kolibri/core/discovery/utils/network/search.py:167]: maybe a good opportunity to switch to new-style `.format` string formatting? (pet peeve of mine, I've always hated `printf`)

Comment (rtibbles) [reply kolibri/core/discovery/utils/network/search.py:167]: Can do more cleanup in the future - yes I dislike it too.

**Review by micahscopes — APPROVED** [review]
Overall this looks great to me.

**Review by indirectlylit — APPROVED** [review]
this seems like an improvement on the previous behavior - I have not noticed the zeroconf unhandled exceptions.

One weird behavior was that it appeared to hang on 'Waiting for child threads to terminate...' - I waited for a few minutes. However then I pressed 'enter' and the process actually seemed like it had finished.

Looks sort of like the behavior described here: https://stackoverflow.com/questions/23125527/how-to-return-to-bash-prompt-after-printing-output-from-backgrounded-function

```
^CINFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
WARNING  SCHEDULECHECKER shut down event received; closing.
INFO     Kolibri instance '211d7b3fd94d8fb8908e8014513318ac' has left the zeroconf network.
WARNING  JOBCHECKER shut down event received; closing.
WARNING  JOBCHECKER shut down event received; closing.
(k013x) d@slate ~/P/l/k013x (release-v0.13.x) [130]> INFO     ENGINE Bus STOPPED
INFO     ENGINE Bus EXITING
INFO     ENGINE Bus EXITED
INFO     ENGINE Caught signal SIGINT.
INFO     ENGINE Bus STOPPING
INFO     ENGINE Bus STOPPED
INFO     ENGINE Bus EXITING
INFO     ENGINE Bus EXITED
INFO     ENGINE Waiting for child threads to terminate...
```

File: kolibri/core/discovery/utils/network/search.py [hunk 3]
```diff
@@ -192,6 +204,7 @@ def unregister_zeroconf_service():
     if ZEROCONF_STATE["service"] is not None:
         ZEROCONF_STATE["service"].cleanup()
     ZEROCONF_STATE["service"] = None
+    ZEROCONF_STATE["zeroconf"].close()
```
Comment (micahscopes) [inline kolibri/core/discovery/utils/network/search.py:207]: who knew :joy_cat:

**Comment by micahscopes** [comment]
> this seems like an improvement on the previous behavior - I have not noticed the zeroconf unhandled exceptions.
> 
> One weird behavior was that it appeared to hang on 'Waiting for child threads to terminate...' - I waited for a few minutes. However then I pressed 'enter' and the process actually seemed like it had finished.
> 
> Looks sort of like the behavior described here: https://stackoverflow.com/questions/23125527/how-to-return-to-bash-prompt-after-printing-output-from-backgrounded-function
> 
> ```
> ^CINFO     ENGINE Caught signal SIGINT.
> INFO     ENGINE Bus STOPPING
> WARNING  SCHEDULECHECKER shut down event received; closing.
> INFO     Kolibri instance '211d7b3fd94d8fb8908e8014513318ac' has left the zeroconf network.
> WARNING  JOBCHECKER shut down event received; closing.
> WARNING  JOBCHECKER shut down event received; closing.
> (k013x) d@slate ~/P/l/k013x (release-v0.13.x) [130]> INFO     ENGINE Bus STOPPED
> INFO     ENGINE Bus EXITING
> INFO     ENGINE Bus EXITED
> INFO     ENGINE Caught signal SIGINT.
> INFO     ENGINE Bus STOPPING
> INFO     ENGINE Bus STOPPED
> INFO     ENGINE Bus EXITING
> INFO     ENGINE Bus EXITED
> INFO     ENGINE Waiting for child threads to terminate...
> ```

@indirectlylit, I've seen that too, it's been around for a few PRs now.  Not really sure what's causing it, but it seems innocuous enough.

**Comment by indirectlylit** [comment]
> Not really sure what's causing it

I assume it's something similar to what I posted

**Comment by indirectlylit** [comment]
I do think it would be nice to fix because based on the final console output, i was literally sitting there for minutes waiting

**Comment by indirectlylit** [comment]
created a follow-up issue here: https://github.com/learningequality/kolibri/issues/6407

not critical for this release

**Pull request merged**